pkgs/by-name/do/docker-init/package.nix

```diff
@@ -0,0 +1,46 @@
+{
+  stdenv,
+  fetchurl,
+  zstd,
+  lib,
+}:
+stdenv.mkDerivation rec {
+  pname = "docker-init";
+  version = "v1.3.0";
+  tag = "157355";
+
+  src = fetchurl {
+    url = "https://desktop.docker.com/linux/main/amd64/${tag}/docker-desktop-x86_64.pkg.tar.zst";
+    hash = "sha256-ysZorPBmoUvTJFFKDbZgQxPamONJKcXezmMrpZSVpwY=";
+  };
+
+  nativeBuildInputs = [
+    zstd
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    tar --zstd -xvf $src
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,libexec/docker/cli-plugins}
+
+    cp usr/lib/docker/cli-plugins/docker-init $out/libexec/docker/cli-plugins
+    ln -s $out/libexec/docker/cli-plugins/docker-init $out/bin/docker-init
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Creates Docker-related starter files for your project";
+    homepage = "https://docs.docker.com/reference/cli/docker/init";
+    downloadPage = "https://docs.docker.com/desktop/release-notes/#4320";
+    mainProgram = "docker-init";
+    license = licenses.unfree;
+    platforms = [ "x86_64-linux" ];
```

Comment:
Could this work on arm linux as well? You could check `stdenv.hostPlatform.isAarch64` and set the `src` appropriately

Reply:
Ah that is a shame, thanks for replying!

Reply:
No problem, thank you for looking at my PR! :)